stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -232,6 +232,15 @@ public struct Unsafe${Mutable}BufferPointer<Element>
 %  end
   }
 
+  /// Creates an empty instance.
+  public init() {
+    // This is known to create a suitably-aligned pointer on all
+    // currently-supported platforms.
```

Comment:
Be warned that the zero page is often populated in embedded systems. Reserving the zero page for anything would hinder future Swift use as a very low-level systems language.